it/UpdateParentLinkFromPayeUnitAcceptanceSpec.scala

```diff
@@ -0,0 +1,230 @@
+        withHeaders(CONTENT_TYPE-> JsonPatchMediaType).
+        patch(s"""[{"op": "replace", "path": "/parents/LEU", "value": "$TargetUBRN"}]"""))
+
+      Then(s"a Unprocessable Entity response is returned")
+      response.status shouldBe UNPROCESSABLE_ENTITY
+    }
+
+    scenario("when the target Legal Unit does not exist in the register") { wsClient =>
+      Given(s"there exists a PAYE unit with reference $PayeRef")
+      stubHBaseFor(aUnitLinksExactRowKeyRequest(withUnitId = PayeUnitId, withUnitType = PayAsYouEarn, withPeriod = RegisterPeriod).
+        willReturn(anOkResponse().withBody(UnitLinksForPayeNoClericalEditsHBaseResponseBody)))
+      And(s"there does not exist a Legal Unit identified by UBRN $TargetUBRN")
+      stubHBaseFor(aUnitLinksExactRowKeyRequest(withUnitId = TargetLegalUnitId, withUnitType = LegalUnit, withPeriod = RegisterPeriod).
+        willReturn(anOkResponse().withBody(NoMatchFoundResponse)))
+
+      When(s"an update of the parent Legal Unit from $IncorrectUBRN to $TargetUBRN is requested for the PAYE unit with reference $PayeRef")
+      val response = await(wsClient.url(s"/v1/periods/${Period.asString(RegisterPeriod)}/types/$PayeUnitAcronym/units/$PayeRef").
+        withHeaders(CONTENT_TYPE -> JsonPatchMediaType).
+        patch(s"""|[{"op": "test", "path": "/parents/LEU", "value": "$IncorrectUBRN"},
+                  | {"op": "replace", "path": "/parents/LEU", "value": "$TargetUBRN"}]""".stripMargin))
+
+      Then(s"a Unprocessable Entity response is returned")
+      response.status shouldBe UNPROCESSABLE_ENTITY
+    }
+
+    scenario("when the PAYE unit being moved does not exist in the register") { wsClient =>
+      Given(s"there does not exist a PAYE unit with reference $PayeRef")
+      stubHBaseFor(aUnitLinksExactRowKeyRequest(withUnitId = PayeUnitId, withUnitType = PayAsYouEarn, withPeriod = RegisterPeriod).
+        willReturn(anOkResponse().withBody(NoMatchFoundResponse)))
+      And(s"there exists a Legal Unit identified by UBRN $TargetUBRN")
+      stubHBaseFor(aUnitLinksExactRowKeyRequest(withUnitId = TargetLegalUnitId, withUnitType = LegalUnit, withPeriod = RegisterPeriod).
+        willReturn(anOkResponse().withBody(UnitLinksForTargetUbrnHBaseResponseBody)))
+
+      When(s"an update of the parent Legal Unit from $IncorrectUBRN to $TargetUBRN is requested for the PAYE unit with reference $PayeRef")
+      val response = await(wsClient.url(s"/v1/periods/${Period.asString(RegisterPeriod)}/types/$PayeUnitAcronym/units/$PayeRef").
+        withHeaders(CONTENT_TYPE-> JsonPatchMediaType).
+        patch(s"""|[{"op": "test", "path": "/parents/LEU", "value": "$IncorrectUBRN"},
+                  | {"op": "replace", "path": "/parents/LEU", "value": "$TargetUBRN"}]""".stripMargin))
+
+      Then(s"a Not Found response is returned")
+      response.status shouldBe NOT_FOUND
+    }
+
+    scenario("when the supplied PAYE reference does not adhere to the expected format") { wsClient =>
```

Comment:
Is this duplication of the routing spec, or is that just for testing the period?

Reply:
It is "duplication" in the sense that it covers a scenario also covered by a unit test, but this is an acceptance test.  This is quite normal.  The unit-test gives the low level, test all eventualities coverage.  The acceptance test covers some of the same examples, but from a more user-focused standpoint.  In this case, the acceptance test attempts to document & cover the API with an example of each major type of issue - but it doesn't cover each and every scenario (we cover invalid paye reference but not invalid period for example - whereas the unit test has both).